src/main/java/com/myhexin/cragent/coderabbittest/FofMqForSendEmailService.java

```diff
@@ -0,0 +1,104 @@
+                        throw new IllegalStateException("#setTradeReq error. unexpected 136+142"
+                                + Parameters.writeAsString("tradeReq", tradeReq));
+                    }
+                    superConvert024To036(tradeReq, splitRef, pdfDto);
+                } else {
+                    if (TradeTypeEnum.SHU_HUI.getCode().equals(splitRef.getBusinessCode())) {
+                        superConvert024To036(tradeReq, splitRef, pdfDto);
+                    } else if (TradeTypeEnum.SHEN_GOU.getCode().equals(splitRef.getBusinessCode())) {
+                        tradeReq.setRemark(String.format(CommonDefine.SUPER_REF_REMARK, tradeReq.getVcAppsheetserialno()));
+                    } else {
+                        throw new IllegalStateException("#setTradeReq error. unexpected 124+122"
+                                + Parameters.writeAsString("tradeReq", tradeReq));
+                    }
+                }
+            } else {
+                tradeReq.setRemark(String.format(CommonDefine.SUPER_REF_REMARK, tradeReq.getVcAppsheetserialno()));
+            }
+        }
+        return tradeReq;
+    }
+
+    public void createPdfDto(Map<String, String> map) {
+        String appDay = tradeDateService.getAppDay().getRespBean();
+        String[] array = map.get("appSheetSerialNo").split(CommonDefine.COMMA);
+        PdfDto pdfDto = new PdfDto();
+        log.info("createPdfDto:map={}", map);
+        // 必要参数校验
+        if (CollectionUtils.isEmpty(map) || StringUtils.isEmpty(map.get("fofId"))
+                || StringUtils.isEmpty(map.get("tradeType")) || StringUtils.isEmpty(map.get("appSheetSerialNo"))) {
+            return;
+        }
+        pdfDto.setDate(DateUtils.getCurrentDay());
+        pdfDto.setCnDate(DateUtils.parseDate(pdfDto.getDate(), DateUtils.DEFAULT_DATE, CommonUtils.CHINESE_DATE));
+        pdfDto.setFofId(map.get("fofId"));
+        pdfDto.setTradeType(map.get("tradeType"));
+        pdfDto.setAppsheetserialnos(map.get("appSheetSerialNo"));
+        TTradeTradereq tradeTradereq = new TTradeTradereq();
+        pdfDto.setTTradeTradereq(tradeTradereq);
+        FofProduct fofProduct = fofProductManager.selectProductByFofId(pdfDto.getFofId());
+        if (null != fofProduct) {
+            pdfDto.setInvestManagerCode(fofProduct.getInvestManagerCode());
+            pdfDto.setOrgId(fofProduct.getOrgId());
+        }
+        // 根据交易码查询对应那些枚举类
+        List<PdfTypeEnum> enums = PdfTypeEnum.getTypeByTradeType(pdfDto.getTradeType());
+        log.info("createPdfDto={}", enums);
+
+        if (CollectionUtils.isEmpty(enums)) {
+            return;
+        }
+        for (PdfTypeEnum pdf : enums) {
+            if (!emailConfigManager.hasSingleSendConfig(pdfDto.getFofId(), pdf.getCode())) {
+                continue;
+            }
+            String tradeCode = map.get("tradeCode");
+            BigDecimal fee = calcFee(tradeCode);
```

Comment:
**Issue**: The method calcFee is called within a loop which might lead to performance issues if the method is computationally expensive. <br> **Fix**: Consider caching the results of calcFee or restructuring the logic to call it less frequently. <br> **Code Suggestion**: 
 ```
 - BigDecimal fee = calcFee(tradeCode);
 + BigDecimal fee = feeCache.computeIfAbsent(tradeCode, this::calcFee);
 ```